request/do.go

```diff
@@ -32,11 +33,22 @@ func doRequest(url string, timeout time.Duration) record.Record {
 // The value of concurrency limits the number of concurrent threads.
 // Once all requests have been made or on done signal from ctx,
 // waits for goroutines to end and returns the collected records.
-func Do(ctx context.Context, requests, concurrency int, url string, timeout time.Duration) []record.Record {
-	records := record.NewSafeSlice(requests)
-
-	semimpl.Do(ctx, concurrency, requests, func() {
-		records.Append(doRequest(url, timeout))
+func Do(cfg config.Config) []record.Record {
+	var (
+		uri        = cfg.Request.URL.String()
+		numWorker  = cfg.RunnerOptions.Concurrency
+		numRequest = cfg.RunnerOptions.Requests
+		reqTimeout = cfg.Request.Timeout
+		gloTimeout = cfg.RunnerOptions.GlobalTimeout
+	)
+
+	records := record.NewSafeSlice(numRequest)
+
+	ctx, cancel := context.WithTimeout(context.Background(), gloTimeout)
+	defer cancel()
```

Comment:
If we were going the go way by passing `context.Context` from the top level process aka main to here, this is a step backward.
Because now the context is not a red thread passed from modules to modules but something one module decide it is.
What was the rationale behind this change ?

Reply:
Nonsense, why would one require to pass the context from main? What I said was using context is a handy pattern specifically designed to achieve what we were doing manually.
Reason is obvious: start the global timout timer closer to when the requests actually starts.